app/models/session.rb

```diff
@@ -11,21 +11,31 @@ def self.check_session_timeout
     purge(ttl)
   end
 
-  def self.purge(ttl)
-    ses = where("updated_at <= ?", ttl.seconds.ago.utc)
-    ses.each do|s|
+  def self.purge(ttl, batch_size = 100)
+    deleted = 0
+    loop do
```

Comment:
I think `find_in_batches` could be the thing (which yields the whole batch to the block).